### DISCUSSION TITLE
Differentiate between a space and a new chord in the On Screen display hint for chained chords.

### DISCUSSION THREAD

**Discussion by Mammal4963** [psoukie/zipchord#222] [Tips and tricks]
![before](https://github.com/user-attachments/assets/f437dd59-9837-45c0-a675-42f70e0d2fe2)
![after](https://github.com/user-attachments/assets/969b2ec0-ccbf-4e77-b9de-e39c9347f9ca)

I changed the "Show Hint" function to the following in the hint.ahk file. This will display a "|" symbol to indicate "new chord" in the on screen hint. This is especially helpful for when child chords contain spaces. It can be difficult for the user to distinguish which child chord gets the space in those circumstances. (see above screenshot example). 

Additionally this brings up a point, we should be checking whether or not there are 3 consecutive spaces in a chord definition before allowing that update to the library. 

ShowHint(line1 := "", line2 := "", line3  := "") {
        global hint_delay
        if (A_Args[1] == "dev") {
            if (test.mode > TEST_STANDBY) {
                test.Log("*Hint*")
            }
            if (test.mode == TEST_RUNNING) {
                return
            }
        }
        hint_delay.Extend()
        if (settings.hints & HINT_TOOLTIP) {
            this._GetCaret(x, y, , h)
            ToolTip % " " . ReplaceWithVariants(line2) . " `n " . ReplaceWithVariants(line3) . " "
                    , x-1.5*h+settings.hint_offset_x, y+1.5*h+settings.hint_offset_y
            hide_tooltip_fn := ObjBindMethod(this, "_HideTooltip")
            SetTimer, %hide_tooltip_fn%, -1800   ; hides the tooltip
        } else {
            if (line2 != ""){
                line2 := StrReplace(line2, " ", "|")
                line2 := StrReplace(line2, "||", "| ")
                if (SubStr(line2, 1, 1) = "|"){
                    line2 := StrReplace(line2, "|", " ", 1)
                }
            }
            this.ShowOnOSD(line1, ReplaceWithVariants(line2, true), ReplaceWithVariants(line3))
        }
    }

**Comment by psoukie** [comment]
Thank you for sharing this! I'll be happy to include this improvement.
